Bump package.json version from 2.4.2 to 2.4.3

SATHISHS03 opened the pull request:
- Updated package.json version from 2.4.1 to 2.4.2.
- Added maxRows and maxCols features to limit the number of rows and columns.
- Updated README.md to reflect the new features .

neSpecc commented:
Why have you merged #165 without 2 approvals? And also, please do not resolve other people's comments. Ask them to review instead. Only author of a comment should resolve it after checking that he has no questions.

neSpecc commented:
Please, fill the PR description

neSpecc commented:
What does it mean "5 by params"? By default there should not be rows and cols limitation

![image](https://github.com/user-attachments/assets/c893f337-04e4-4de8-969d-29d0d2c18120)

SATHISHS03 commented:
> What does it mean "5 by params"? By default there should not be any rows and cols limitations
> 
> ![image](https://private-user-images.githubusercontent.com/3684889/370711371-c893f337-04e4-4de8-969d-29d0d2c18120.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcyNzI2OTIsIm5iZiI6MTcyNzI3MjM5MiwicGF0aCI6Ii8zNjg0ODg5LzM3MDcxMTM3MS1jODkzZjMzNy0wNGU0LTRkZTgtOTY5ZC0yOWQwZDJjMTgxMjAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjVUMTM1MzEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzI1ZTA1OWM5YTgyYTJkMTEzODI3OGVkNmY1MTk4ZDQ5MWRlMzVmYjk5OTQ5ZjhhOThlZTIxZGFiNjczZjU3YSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.wp20_SVYBq0-qIMQEGrBsLOGnAMHsrReGrVRvqY-reg)


Since I've added maxRows: 5 and maxCols: 5 to both config and index.html, the table now displays only 5 rows and 5 columns when these parameters are passed as an example ,.

Should I remove or modify them?

neSpecc commented:
> > What does it mean "5 by params"? By default there should not be any rows and cols limitations

> Since I've added maxRows: 5 and maxCols: 5 to both config and index.html, the table now displays only 5 rows and 5 columns when these parameters are passed as an example ,.
> 
> Should I remove or modify them?

I still did't get what does it mean "by params" in README. Which params? You need to update the Readme making it more clear for developers.

neSpecc commented:
Another think is code duplication. You have many similar fragments like these:

![image](https://github.com/user-attachments/assets/17df26c0-639b-4c89-aedf-ab391c7c64aa)

Instead of adding it in many places, create a method `updateAddColumnButtonState()` and simply call it where needed

SATHISHS03 commented:
> Another think is code duplication. You have many similar fragments like these:
> 
> ![image](https://private-user-images.githubusercontent.com/3684889/370827456-17df26c0-639b-4c89-aedf-ab391c7c64aa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc1MzgwNjcsIm5iZiI6MTcyNzUzNzc2NywicGF0aCI6Ii8zNjg0ODg5LzM3MDgyNzQ1Ni0xN2RmMjZjMC02MzliLTRjODktYWVkZi1hYjM5MWM3YzY0YWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjhUMTUzNjA3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODc3MWNiOWQwN2M0NzA0N2U0M2YwZjM4MjBiNzhjMTdlMmI0NGJlNTgyZmM4OTUwY2NiYzE3NDZjMzdhYmI3YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.v00z2flh0nEWv7ZlN6nuT41_ZJVJ0hSkx4t9DUYfZo8)
> 
> Instead of adding it in many places, create a method `updateAddColumnButtonState()` and simply call it where needed

created updateAddButtonstate , handles repeated logic

e11sy commented on README.md:
```diff
@@ -61,8 +61,8 @@ var editor = EditorJS({
 | `cols`             | `number` | initial number of columns. `2` by default |
-| `maxRows`          | `number` | maximum number of rows. `5` by params |
-| `maxCols`          | `number` | maximum number of columns. `5` by params |
+| maxRows            | `number` | Sets the maximum number of rows allowed in the table. `Defaults` to `undefined` (no limit) |
+| `maxCols`          | `number` | Sets the maximum number of columns allowed in the table. Defaults to `undefined` (no limit) |
```
```suggestion
| `maxRows`        | `number` | Sets the maximum number of rows allowed in the table. `undefined` by default (no limit set) |
| `maxCols`          | `number` | Sets the maximum number of columns allowed in the table. `undefined` by default (no limit set)  |
```

e11sy commented on src/table.js:
```diff
@@ -353,6 +353,25 @@ export default class Table {
+  updateButtonState(type) {
+    const maxLimit = (type =="Row") ? this.config.maxRows : this.config.maxCols;
+    const currentCount = (type == "Row") ? this.numberOfRows : this.numberOfColumns;
+    const addButton =(type == "Row") ?this.wrapper.querySelector(`.${CSS.addRow}`) : this.wrapper.querySelector(`.${CSS.addColumn}`);
```
```suggestion
    const addButton = (type == "Row") ? this.wrapper.querySelector(`.${CSS.addRow}`) :  this.wrapper.querySelector(`.${CSS.addColumn}`);
```

e11sy commented on src/table.js:
```diff
@@ -353,6 +353,25 @@ export default class Table {
+    const currentCount = (type == "Row") ? this.numberOfRows : this.numberOfColumns;
+    const addButton =(type == "Row") ?this.wrapper.querySelector(`.${CSS.addRow}`) : this.wrapper.querySelector(`.${CSS.addColumn}`);
+    if (addButton) {
+      if (maxLimit && currentCount >= maxLimit ) {
```
```suggestion
      if (maxLimit && currentCount >= maxLimit) {
```

SATHISHS03 commented:
> > > What does it mean "5 by params"? By default there should not be any rows and cols limitations
> 
> > Since I've added maxRows: 5 and maxCols: 5 to both config and index.html, the table now displays only 5 rows and 5 columns when these parameters are passed as an example ,.
> > Should I remove or modify them?
> 
> I still did't get what does it mean "by params" in README. Which params? You need to update the Readme making it more clear for developers. Mention that they are optional params.

updated readme.md , removed params , made it more clear

neSpecc commented on src/table.js:
```diff
@@ -353,6 +353,25 @@ export default class Table {
   }
 
+  /**
+  * Update the state of the add button based on the current count and maximum allowed count.
```
```suggestion
  * Update the state of the "Add Column" or "Add Row" button based on the current rows/cols count and maximum allowed count.
```

neSpecc commented on src/table.js:
```diff
@@ -353,6 +353,25 @@ export default class Table {
+  * @param {'Row'|'Column'} type - The type of element ('Row' or 'Column') to update.
+  */
+  updateButtonState(type) {
+    const maxLimit = (type =="Row") ? this.config.maxRows : this.config.maxCols;
+    const currentCount = (type == "Row") ? this.numberOfRows : this.numberOfColumns;
+    const addButton = (type == "Row") ? this.wrapper.querySelector(`.${CSS.addRow}`) :  this.wrapper.querySelector(`.${CSS.addColumn}`);
```
redundant parentheses

neSpecc commented on src/table.js:
```diff
@@ -475,10 +488,8 @@ export default class Table {
-      addColButton.classList.remove(CSS.addColumnDisabled);
-    }
+    
+    this.updateButtonState("Column");
```
Seems like redundant operation.

I think, we can update buttons state right before opening the toolbox.

SATHISHS03 replied:
Can you give me bit more information on that  .. how can i eliminate the reductant operation

neSpecc replied:
every time Block Tunes menu is open, `renderSettings()` is called. So you don't need to update its buttons state when it is close. Just make sure to get correct state when opening

neSpecc commented on src/table.js:
```diff
@@ -488,11 +499,7 @@ export default class Table {
-      addRowButton.classList.remove(CSS.addRowDisabled);
-    }
-
+    this.updateButtonState("Row");
```
same

SATHISHS03 replied:
- **Optimization:** Eliminated unnecessary calls to `updateButtonState` during table rendering. Previously, the `updateButtonState` function was being called from within the `addRow`, `addColumn`, `deleteRow`, and `deleteColumn` methods, causing multiple redundant updates during rendering.
- **Improvement:** Now, `updateButtonState` is only called directly in the click event handlers for adding or deleting rows and columns. This ensures that the button state is updated only when user actions (like clicking 'Add' or 'Delete') are performed, leading to better performance and reduced redundant function calls.

e11sy reviewed (commented):
now actual version is `2.4.2`, so bumped would be `2.4.3`

neSpecc reviewed (changes requested):
@SATHISHS03 you don't need to update Tunes when there are hidden. Do it only in `renderSettings`
